### PR TITLE
Check if request is aborted before verifying response bytes written (#1498)

### PR DESCRIPTION
**Pull request by cesarbs**
#1498

### CODE REVIEW

File: src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs [hunk 1]
```diff
@@ -94,7 +94,11 @@ public Frame(IHttpApplication<TContext> application, ConnectionContext context)
                             try
                             {
                                 await _application.ProcessRequestAsync(context).ConfigureAwait(false);
-                                VerifyResponseContentLength();
+
+                                if (Volatile.Read(ref _requestAborted) == 0)
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs]: Is this set for both dead sockets and anything that's called Abort?

Comment (cesarbs) [inline src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs]: Yes.

**Comment by cesarbs** [comment]
Checking Travis failure.

**Comment by davidfowl** [comment]
@cesarbs file a bug for this failure. After all of this churn we need to make sure the tests are stable again.

File: test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs [hunk 2]
```diff
@@ -659,12 +662,55 @@ public Task ResponseStatusCodeSetBeforeHttpContextDisposedRequestMalformedReadIg
...
+            {
+                httpContext.RequestAborted.Register(() =>
+                {
+                    abortTcs.SetResult(null);
+                });
+                httpContext.Response.ContentLength = 12;
+                await httpContext.Response.WriteAsync("hello,");
+
+                // Wait until request is aborted so we know Frame will skip the response content length check.
+                await abortTcs.Task.TimeoutAfter(TimeSpan.FromSeconds(10));
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Would this test still pass of this timed out after 10s without being aborted? If so, make it so it fails if the request doesn't get aborted. I hat tests that time out and silently pass.

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.Kestrel.FunctionalTests/ResponseTests.cs]: Good catch. Updated.

**Pull request merged**